cloudinit/sources/DataSourceScaleway.py

```diff
@@ -249,7 +259,7 @@ def network_config(self):
 
         netcfg = {"type": "physical", "name": "%s" % self._fallback_interface}
         subnets = [{"type": "dhcp4"}]
-        if self.metadata["ipv6"]:
+        if 'ipv6' in self.metadata:
```

Comment:
The following changeset is wrong. The test is not equivalent :
 - original test tests the value set for the 'ipv6' key
 - changed test verify that the key exists

The 'ipv6' key may exist with a None value. As a result of that, the three(3) following unittests are failing :

  - python3 -m pytest -v tests/unittests/sources test_scaleway.py::TestDataSourceScaleway::test_network_config_cached_none
  - python3 -m pytest -v tests/unittests/sources/test_scaleway.py::TestDataSourceScaleway::test_network_config_ok
  - python3 -m pytest -v tests/unittests/sources/test_scaleway.py::TestDataSourceScaleway::test_network_config_unset

Reply:
right, then you can use : 
if 'ipv6' in self.metadata:
    if self.metadata["ipv6"]:
    ...
which is easier to read

Reply:
I'm agree with it